diff --git a/qtbase/mkspecs/macx-ios-clang/qmake.conf b/qtbase/mkspecs/macx-ios-clang/qmake.conf
index d58b9fc..221c66c 100644
--- a/qtbase/mkspecs/macx-ios-clang/qmake.conf
+++ b/qtbase/mkspecs/macx-ios-clang/qmake.conf
@@ -2,13 +2,13 @@
 # qmake configuration for macx-ios-clang
 #
 
-QMAKE_IOS_DEPLOYMENT_TARGET = 10.0
+QMAKE_IOS_DEPLOYMENT_TARGET = 9.0
 
 # Universal target (iPhone and iPad)
 QMAKE_APPLE_TARGETED_DEVICE_FAMILY = 1,2
 
-QMAKE_APPLE_DEVICE_ARCHS = arm64
-QMAKE_APPLE_SIMULATOR_ARCHS = x86_64
+QMAKE_APPLE_DEVICE_ARCHS = armv7 arm64
+QMAKE_APPLE_SIMULATOR_ARCHS = i386 x86_64
 
 include(../common/ios.conf)
 include(../common/gcc-base-mac.conf)
